actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -128,9 +128,19 @@ def initialize(set:, ast:, controller:, default_action:, to:, formatted:, via:,
           @internal           = options.delete(:internal)
           @scope_options      = scope_params[:options]
 
-          path_params = ast.find_all(&:symbol?).map(&:to_sym)
+          path_params = []
+          wildcard_options = {}
+          ast.each do |node|
+            if node.symbol?
+              path_params << node.to_sym
+            elsif node.star? && formatted != false
+              # Add a constraint for wildcard route to make it non-greedy and match the
+              # optional format part of the route by default.
+              wildcard_options[node.name.to_sym] ||= /.+?/
```

Comment:
can a `node`  ever have a `nil` name? I was thinking safe-navigation would be safer in this case. Feel free to ignore of course as I have no idea what this `node` obj is and how its formed.

Reply:
No, I don't believe it can be `nil`. `node.name.to_sym` was taken from the previous code, which has been around for some years now, so I think we are good. 👍